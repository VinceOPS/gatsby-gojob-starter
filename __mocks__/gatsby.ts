const gatsby = jest.requireActual('gatsby');

// hack: avoid mocking <Link> component from gatsby
// eslint-disable-next-line @typescript-eslint/no-explicit-any
(window as any).___loader = { enqueue: jest.fn() };

module.exports = {
  ...gatsby,
  graphql: jest.fn(),
  StaticQuery: jest.fn(),
  useStaticQuery: jest.fn(),
};
