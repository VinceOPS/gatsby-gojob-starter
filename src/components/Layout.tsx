import React, { FunctionComponent } from 'react';
import PropTypes from 'prop-types';
import { graphql, useStaticQuery } from 'gatsby';

import Header from './gatsby/Header';
import './layout.css';

/**
 * Layout component that queries for data with Gatsby's useStaticQuery React hook
 * @see https://www.gatsbyjs.org/docs/use-static-query/
 */
const Layout: FunctionComponent = ({ children }) => {
  const { site } = useStaticQuery(
    graphql`
      query SiteTitleQuery {
        site {
          siteMetadata {
            title
          }
        }
      }
    `,
  );

  return (
    <>
      <Header siteTitle={site.siteMetadata.title} />
      <div
        style={{
          margin: '0 auto',
          maxWidth: 960,
          padding: '0px 1.0875rem 1.45rem',
          paddingTop: 0,
        }}
      >
        <main>{children}</main>
        <footer>
          Â© {new Date().getFullYear()}, Built with <a href="https://www.gatsbyjs.org">Gatsby</a>
        </footer>
      </div>
    </>
  );
};

Layout.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Layout;
