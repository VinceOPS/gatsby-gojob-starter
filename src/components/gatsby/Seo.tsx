import React, { FunctionComponent } from 'react';
import PropTypes from 'prop-types';
import Helmet from 'react-helmet';
import { useStaticQuery, graphql } from 'gatsby';

type SeoProps = {
  description?: string;
  lang?: string;
  meta?: { name?: string; content?: string; property?: string }[];
  keywords?: string[];
  title?: string;
};

/**
 * SEO component that queries for data with Gatsby's useStaticQuery React hook
 * @see https://www.gatsbyjs.org/docs/use-static-query/
 */
const Seo: FunctionComponent<SeoProps> = ({ description, lang, meta = [], keywords = [], title }) => {
  const { site } = useStaticQuery(
    graphql`
      query {
        site {
          siteMetadata {
            title
            description
            author
          }
        }
      }
    `,
  );

  const metaDescription = description || site.siteMetadata.description;

  return (
    <Helmet
      htmlAttributes={{
        lang,
      }}
      title={title}
      titleTemplate={`%s | ${site.siteMetadata.title}`}
      meta={[
        {
          name: 'description',
          content: metaDescription,
        },
        {
          property: 'og:title',
          content: title,
        },
        {
          property: 'og:description',
          content: metaDescription,
        },
        {
          property: 'og:type',
          content: 'website',
        },
        {
          name: 'twitter:card',
          content: 'summary',
        },
        {
          name: 'twitter:creator',
          content: site.siteMetadata.author,
        },
        {
          name: 'twitter:title',
          content: title,
        },
        {
          name: 'twitter:description',
          content: metaDescription,
        },
        ...(keywords.length > 0 ? [{ name: 'keywords', content: keywords.join(', ') }] : []),
        ...meta,
      ]}
    />
  );
};

Seo.defaultProps = {
  lang: 'en',
  meta: [],
  keywords: [],
  description: '',
};

Seo.propTypes = {
  description: PropTypes.string,
  lang: PropTypes.string,
  meta: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      content: PropTypes.string,
      property: PropTypes.string,
    }).isRequired,
  ),
  keywords: PropTypes.arrayOf(PropTypes.string.isRequired),
  title: PropTypes.string.isRequired,
};

export default Seo;
