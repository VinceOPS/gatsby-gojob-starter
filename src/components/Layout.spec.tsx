import React from 'react';
import 'jest-dom/extend-expect';
import { render } from 'react-testing-library';

import Layout from './Layout';
import { useStaticQuery } from 'gatsby';

describe('<Layout />', () => {
  const title = 'My page being tested';
  const content = 'Hello World';
  const children = <div>{content}</div>;

  beforeEach(() => {
    (useStaticQuery as jest.Mock).mockReturnValue({ site: { siteMetadata: { title } } });
  });

  it('contains the title of the website', () => {
    const { container } = render(<Layout>{children}</Layout>);
    expect(container).toHaveTextContent(title);
  });

  it('contains the text of its children', () => {
    const { container } = render(<Layout>{children}</Layout>);
    expect(container).toHaveTextContent(content);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Layout>{children}</Layout>);
    expect(asFragment()).toMatchSnapshot();
  });
});
