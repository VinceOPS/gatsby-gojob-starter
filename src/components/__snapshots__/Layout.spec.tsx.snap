// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> matches snapshot 1`] = `
<DocumentFragment>
  <header
    style="background: rebeccapurple; margin-bottom: 1.45rem;"
  >
    <div
      style="margin: 0px auto; max-width: 960px; padding: 1.45rem 1.0875rem;"
    >
      <h1
        style="margin: 0px;"
      >
        <a
          aria-current="page"
          class=""
          href="/"
          style="color: white; text-decoration: none;"
        >
          My page being tested
        </a>
      </h1>
    </div>
  </header>
  <div
    style="margin: 0px auto; max-width: 960px; padding: 0px 1.0875rem 1.45rem 1.0875rem;"
  >
    <main>
      <div>
        Hello World
      </div>
    </main>
    <footer>
      Â© 2019, Built with 
      <a
        href="https://www.gatsbyjs.org"
      >
        Gatsby
      </a>
    </footer>
  </div>
</DocumentFragment>
`;
